crates/subspace-farmer/src/farming.rs

```diff
@@ -188,19 +191,40 @@ async fn subscribe_to_slot_info<T: RpcClient>(
 
                 // Try to first find a block authoring solution, then if not found try to find a
                 // vote
-                let maybe_tag = commitments
-                    .find_by_range(target, slot_info.solution_range, slot_info.salt)
-                    .or_else(|| {
-                        if slot_info.solution_range == slot_info.voting_solution_range {
-                            return None;
-                        }
-
-                        commitments.find_by_range(
-                            target,
-                            slot_info.voting_solution_range,
-                            slot_info.salt,
-                        )
-                    });
+                let voting_tags = commitments.find_by_range(
+                    target,
+                    slot_info.voting_solution_range,
+                    slot_info.salt,
+                    TAGS_SEARCH_LIMIT,
+                );
+
+                let maybe_tag = if voting_tags.len() < TAGS_SEARCH_LIMIT {
+                    // We found all tags within voting solution range
+                    voting_tags.into_iter().next()
```

Comment:
Don't we want to check `is_within_solution_range` here too? In this case there were less than 10 tags found in `voting_solution_range`, but there's still a possibility some of them fall into `solution_range`?

Reply:
`voting_solution_range` is never smaller than `solution_range`. If we requested `TAGS_SEARCH_LIMIT` tags and found less, we have clearly exhausted the whole range of `voting_solution_range` to do so and thus already checked `solution_range` too, so there is no point of checking anything else here, just return the closest tag to the target.